packages/schematics/angular/utility/scripts.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Tree } from '@angular-devkit/schematics';
+import {
+  appendPropertyInAstObject,
+  findPropertyInAstObject,
+  insertPropertyInAstObjectInOrder,
+} from './json-utils';
+import { pkgJsonPath, readPackageJson } from './package-json';
+
+export interface NodeScript {
+  name: string;
+  script: string;
+}
+
+export function addPackageJsonScript(tree: Tree, script: NodeScript): void {
+  const packageJsonAst = readPackageJson(tree);
+  const scriptsNode = findPropertyInAstObject(packageJsonAst, 'scripts');
+  const recorder = tree.beginUpdate(pkgJsonPath);
+  if (!scriptsNode) {
+    // Haven't found the scripts key, add it to the root of the package.json.
+    appendPropertyInAstObject(recorder, packageJsonAst, 'scripts', {
+      [script.name]: script.script,
+    }, 2);
+  } else if (scriptsNode.kind === 'object') {
+    // check if script already added
+    const scriptNode = findPropertyInAstObject(scriptsNode, script.name);
+
+    if (!scriptNode) {
```

Comment:
This is taken from the dependencies utilities.
Except it doesn't overwrite the script value if it's already added.